test/tools/cmap_spec_runner.ts

```diff
@@ -0,0 +1,338 @@
+import { expect } from 'chai';
+import { EventEmitter } from 'events';
+import * as util from 'util';
+
+import { Connection, HostAddress } from '../../src';
+import { ConnectionPool } from '../../src/cmap/connection_pool';
+import { sleep } from './utils';
+
+type CmapOperation =
```

Comment:
just a question - do we have a practice of when we use interfaces and when we use types?

Reply:
I usually think of types as more static (unchanging) than interfaces, but [functionally](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces) they're more or less equivalent in most cases and in this one specifically. @nbbeeken might have a stronger opinion here

Reply:
I agree that here it's equivalent, and we don't need to ever `extends` these so type does the job.

From the linked docs though I do like this feature of interfaces: 
- Interface names will [always appear in their original form](https://www.typescriptlang.org/play?#code/PTAEGEHsFsAcEsA2BTATqNrLusgzngIYDm+oA7koqIYuYQJ56gCueyoAUCKAC4AWHAHaFcoSADMaQ0PCG80EwgGNkALk6c5C1EtWgAsqOi1QAb06groEbjWg8vVHOKcAvpokshy3vEgyyMr8kEbQJogAFND2YREAlOaW1soBeJAoAHSIkMTRmbbI8e6aPMiZxJmgACqCGKhY6ABGyDnkFFQ0dIzMbBwCwqIccabcYLyQoKjIEmh8kwN8DLAc5PzwwbLMyAAeK77IACYaQSEjUWY2Q-YAjABMAMwALA+gbsVjNXW8yxySoAADaAA0CCaZbPh1XYqXgOIY0ZgmcK0AA0nyaLFhhGY8F4AHJmEJILCWsgZId4NNfIgGFdcIcUTVfgBlZTOWC8T7kAJ42G4eT+GS42QyRaYbCgXAEEguTzeXyCjDBSAAQSE8Ai0Xsl0K9kcziExDeiQs1lAqSE6SyOTy0AKQ2KHk4p1V6s1OuuoHuzwArMagA) in error messages, but only when they are used by name.

When we do type checking for tests or just in general this is a good thing to keep in mind for UX. But no need to make a change here.

Reply:
Named types also appear in error messages when used by name too 😃 ([playground](https://www.typescriptlang.org/play?#code/PTAEGEHsFsAcEsA2BTATqNrLusgzngIYDm+oA7koqIYuYQJ56gCueyoAUCKAC4AWHAHaFcoSADMaQ0PCG80EwgGNkALk6deDWBwCyo6LVABeUAG9Ooa6BG41oPL1RzinAL6aJLIct7xIGWRlfkgDaCNEAApoB3DIgEoLKxtlQLxIFAA6REhiGKy7ZASPTR5kLOIs0AAVQQxULHQAI2Rc8goqGjpGZjYOAWFRfUNjbjBeSFBUZAk0PinBvh0Ocn54ENlmZAAPXT9kABMNYND42ijzW2GHAEYAJgBmABZH0HcS8dr67V1xKQABtAAaBBDMtnx6rsVLxHMMaMwjBFaAAaL7NFiwwjMeC8ADkzCEkFhrWQMkO8BmfkQDGuuEOKNqKwAysoXLBeF9yIE8bDcPIAjJcbIZEtMNhQLgCCRXF4fH5BRgQpAAIJCeCRGIOK5FBxOFxCYjvJKWGygNJCDLZXL5aCFYYlTycU6q9WanU3UAPF4AVmNQA))

Agreed that there's no need to make a change here so I'll resolve this conversation